### PR TITLE
feat: add expiry to long living api tokens

### LINKED ISSUE #6964: Add an optional expiration date on api token for the wallet

### PR DESCRIPTION
**Pull request by jeremyletang**
close #6964

### CODE REVIEW

File: wallet/api/services_manager.go [hunk 1]
```diff
@@ -49,7 +51,13 @@ func (ns *ServicesManager) RegisterService(network, url string, sessions *sessio
 		if err != nil {
 			return fmt.Errorf("could not retrieve the wallet %q associated to the token %q: %w", tokenInfo.Wallet.Name, token.Token, err)
 		}
-		if err := sessions.ConnectWalletForLongLivingConnection(token.Token, w); err != nil {
+
+		var expiry *time.Time
+		if tokenInfo.Expiry != nil {
+			expiry = ptr.From(time.Unix(*tokenInfo.Expiry, 0))
+		}
```
Comment (ValentinTrinque) [inline wallet/api/services_manager.go:58]: This check should be delegated to the `sessions.ConnectWalletForLongLivingConnection()` as it's its responsability to check valid input, _ultimately_.

If we are to call `sessions.ConnectWalletForLongLivingConnection()` somewhere else we don't want to copy the expiry verification.

File: wallet/api/session/sessions.go [hunk 2]
```diff
@@ -13,6 +14,7 @@ var (
 	ErrNoWalletConnected                       = errors.New("no wallet connected")
 	ErrCannotEndLongLivingSessions             = errors.New("sessions attached to long-living tokens cannot be ended")
 	ErrGeneratedTokenCollidesWithExistingToken = errors.New("the generated token collides with an existing token")
+	ErrAPITokenExpired                         = errors.New("api token expired")
```
Comment (ValentinTrinque) [inline wallet/api/session/sessions.go:17]: It is user facing, we need something very user friendly

```suggestion
	ErrAPITokenExpired                         = errors.New("the token has expired")
```

File: wallet/api/session/store/v1/file_store.go [hunk 3]
```diff
@@ -11,7 +11,10 @@ import (
 	"code.vegaprotocol.io/vega/wallet/api/session"
 )
 
-var ErrStoreNotInitialized = errors.New("the tokens store has not been initialized")
+var (
+	ErrStoreNotInitialized = errors.New("the tokens store has not been initialized")
+	ErrTokenExpired        = errors.New("token expired")
```
Comment (ValentinTrinque) [inline wallet/api/session/store/v1/file_store.go:16]: Is this used ?

**Review by ValentinTrinque — CHANGES_REQUESTED** [review]

File: wallet/api/client_sign_transaction.go [hunk 4]
```diff
@@ -157,11 +158,21 @@ func (h *ClientSignTransaction) Handle(ctx context.Context, rawParams jsonrpc.Pa
...
-func NewSignTransaction(interactor Interactor, nodeSelector node.Selector, sessions *session.Sessions) *ClientSignTransaction {
+func NewSignTransaction(interactor Interactor, nodeSelector node.Selector, sessions *session.Sessions, tp ...TimeProvider) *ClientSignTransaction {
+	if len(tp) > 1 {
+		panic("only one time provider allowed at most")
+	}
+
+	var t TimeProvider = &StdTime{}
+	if len(tp) > 0 {
+		t = tp[0]
+	}
```
[the remark this answers is not in the document]
Comment (jeremyletang) [reply wallet/api/client_sign_transaction.go:169]: Well, I just liked better having a default.

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**